openshift_scalability/content/centos-stress/README.md

```diff
@@ -1,2 +1,95 @@
+
+OCP nodes hosting WLG pods must be labeled by a `placement=${PLACEMENT}`
+label, otherwise the WLG pods will fail to fit any node.  The default value of
+`${PLACEMENT}` in [stress-pod.json](https://github.com/openshift/svt/blob/master/openshift_scalability/content/quickstarts/stress/stress-pod.json)
+is `test`.
+
+Before running any of the test harnesses synchronised by the cluster-loader [autogen functionality](https://github.com/openshift/svt/blob/master/openshift_scalability/README.md) ensure you can connect from OCP nodes hosting WLG pods to a machine running cluster-loader (OCP master) to TCP port 9090.  This value can be changed by overriding GUN_PORT variable (see
+[stress-pod.json](https://github.com/openshift/svt/blob/master/openshift_scalability/content/quickstarts/stress/stress-pod.json)).
+
+## Test harnesses
+
+There are several test harnesses provided by this docker image.  The recommended
+way of running them is through the use of the [cluster-loader](https://github.com/openshift/svt/blob/master/openshift_scalability/cluster-loader.py) tool.  All of the test harnesses use a common template
+[stress-pod.json](https://github.com/openshift/svt/blob/master/openshift_scalability/content/quickstarts/stress/stress-pod.json), but they typically use their own cluster-loader configuration file.
+
+### stress
+
+Add description.
+
+### JMeter
+
+Add description.
+
+### wrk
+
+This test harness makes use of a modern HTTP benchmarking tool [wrk](https://github.com/wg/wrk) with support for scripting in LUA. [stress-wrk.yaml](https://github.com/openshift/svt/blob/master/openshift_scalability/config/stress-wrk.yaml) is an example of the cluster-loader configuration file.
+
+#### System tuning
+
+When using the HTTP no keep-alive mode (`WRK_KEEPALIVE` set to `n` in [stress-wrk.yaml](https://github.com/openshift/svt/blob/master/openshift_scalability/config/stress-wrk.yaml)) it is necessary to set kernel configuration
+`net.ipv4.tcp_tw_reuse=1` on WLG/wrk client nodes, otherwise the client system
+will quickly run out of usable sockets.
+
+#### Performance and limitations
+
+The [wrk](https://github.com/wg/wrk) client was designed to put a significant
+amount of load on a single HTTP(S) server.  An optional LuaJIT script can
+perform HTTP request generation, response processing, and custom reporting, but
+this is at the expense of load it can generate.  However, the amount of load
+generated in this way is still very high.  With a
+[patch](https://github.com/openshift/svt/blob/master/utils/wrk/patches/wrk-4.0.2.diff)
+to allow targeting multiple
+hosts from LUA scripts, a wrk WLG pod is capable of sending and processing around 30000 HTTP requests per second evenly spread across 2000 routes (i.e. around 15 rps per single route) on an EC2 m4.xlarge VM instance.  Using more than 2000 routes is not advisable.  Note
+that increasing the number of clients/connections per thread (WRK_CONNS_PER_THREAD)
```

Comment:
At around 2500 threads/routes I was starting to run out of virtual memory and the client was crashing because of that.  There was too much of context switching and the load on the client machine was just too high.